### PR TITLE
Support non-Latin paths in i_glob.c, don't use SDL_iconv_string()

### PR DESCRIPTION
**Pull request by rfomin**

### CODE REVIEW

File: src/i_glob.c [hunk 1]
```diff
@@ -140,15 +143,15 @@ glob_t *I_StartMultiGlob(const char *directory, int flags,
         return NULL;
     }
 
-    result->dir = opendir(directory);
+    result->dir = opendir(directory_native);
```
[the remark this answers is not in the document]
Comment (fabiangreffrath) [reply src/i_glob.c:146]: Yes, but we return early here.

[the remark this answers is not in the document]
Comment (rfomin) [reply src/i_glob.c:146]: Right, fixed in https://github.com/chocolate-doom/chocolate-doom/pull/1557/commits/6b4babf7f84f80bf68a6f18ffe5bc77b4a078a85

File: src/m_misc.h [hunk 2]
```diff
@@ -29,6 +29,8 @@
 #ifdef _WIN32
 wchar_t *M_ConvertUtf8ToWide(const char *str);
 #endif
+char *M_ConvertUtf8ToSysNativeMB(const char *str);
```
Comment (turol) [inline src/m_misc.h:32]: There should be a comment here that these return a new string which must be `free()`d by the caller.

Comment (rfomin) [reply src/m_misc.h:32]: Fixed in https://github.com/chocolate-doom/chocolate-doom/pull/1557/commits/aca14b69c1e4bc527118ead32a7a5cc6d32334e9

**Review by turol — APPROVED** [review]

**Pull request merged**